flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroSerializationSchema.java

```diff
@@ -220,4 +226,22 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(recordClazz, schema);
     }
+
+    /**
+     * Returns the type information of the produced type. Depending on the type of the record,
+     * whether it is a specific or generic record, the type information will be different. It can be
+     * either a {@link AvroTypeInfo} or a {@link GenericRecordAvroTypeInfo}.
+     *
+     * @return TypeInformation of the produced type
+     */
+    @Override
+    public TypeInformation getProducedType() {
```

Comment:
This should be `TypeInformation<T>`. You probably need to cast the returns though but I think it makes things more explicit.

Reply:
The common type for both cases is `ResultTypeQueryable<GenericContainer>` as both `SpecificRecord` and `GenericRecord` extend `GenericContainer`. However, I could not make it work for `TypeInformation<GenericRecord>` to be casted into `TypeInformation<GenericContainer>`